.travis.yml

```diff
@@ -1,13 +1,18 @@
 language: python
 python:
     - "2.7"
-    - "3.2"
+    - "3.7"
+cache: pip
 before_install:
  - sudo apt-get update -qq
  - sudo apt-get install -qq --no-install-recommends ruby-sass
 install:
-    - pip install nose
-    - pip install -e git://github.com/getpelican/pelican.git#egg=pelican
-    - pip install --use-mirrors Markdown
-    - if [[ $TRAVIS_PYTHON_VERSION == '2.7' ]]; then pip install --use-mirrors webassets cssmin; fi
-script: nosetests
+    - pip install -r requirements.txt
+    # Installing pelican in editable mode makes its tests.support package available to us
+    - pip install -e git+https://github.com/getpelican/pelican.git#egg=pelican
+    # The following commands install all plugins dependencies.
+    # The `rm`, `grep` & `sed commands` below are needed because some submodules have conflicting requirements.
+    # Also, Cython is required for the plugins pelican-ipynb through jupyter & panorama through pandas.
```

Comment:
Why does `Cython` need to be explicit? Doesn't it resolve through `jupyter` or `pandas`?

Reply:
My mistake, this is not needed indeed

Reply:
Without it, this is the kind of errors I get:
https://travis-ci.org/Lucas-C/pelican-plugins/jobs/580703256
```
   Exception: Cython-generated file 'pandas/lib.c' not found.
                    Cython is required to compile pandas from a development branch.
                    Please install Cython or download a release package of pandas.
```

Reply:
My mistake, this is not needed indeed
This error was due to the locked PyYaml dependency from the `panorama` plugin